pkgs/development/python-modules/jax/default.nix

```diff
@@ -61,13 +62,18 @@ buildPythonPackage rec {
     jaxlib'
     matplotlib
     pytestCheckHook
+    pytest-xdist
   ];
 
+  # don't use max jobs
+  dontUsePytestXdist = true;
+
   # NOTE: Don't run the tests in the expiremental directory as they require flax
   # which creates a circular dependency. See https://discourse.nixos.org/t/how-to-nix-ify-python-packages-with-circular-dependencies/14648/2.
   # Not a big deal, this is how the JAX docs suggest running the test suite
   # anyhow.
   pytestFlagsArray = [
+    "--numprocesses=3"
```

Comment:
Based on your [comment](https://github.com/NixOS/nixpkgs/issues/258089#issuecomment-1741587229) earlier, this will require a peak memory usage of 15G. We need to keep the memory usage low enough to run the build and test suite on a GH Actions free runner for nixpkgs-upkeep. Unfortunately those runners have 7G of memory. So we'll have to go lower -- what's the peak memory usage of `"--numprocesses=2"`?

Reply:
With two processes I'm still seeing ~13.2 GB of peak memory usage.

Reply:
Ok, let's keep the test suite single-threaded and then just fix the AMD CPU issue in this PR. We can always brainstorm what to do about pytest-xdist in the future.